client/state/plugins/installed/actions.js

```diff
@@ -607,12 +615,12 @@ export function fetchAllPlugins() {
 		const receivePluginsDispatchSuccess = ( { sites } ) => {
 			dispatch( { type: PLUGINS_ALL_REQUEST_SUCCESS } );
 
+			dispatch( receiveAllSitesPlugins( sites ) );
+
 			Object.entries( sites ).forEach( ( [ siteId, plugins ] ) => {
 				// Cast the enumerable string-keyed property to a number.
 				siteId = Number( siteId );
 
-				dispatch( receiveSitePlugins( siteId, plugins ) );
```

Comment:
Loving this improvement ❤️ It improves the performance and user experience in such a visible way!